From f537cf268c8915dc97b269976bdc6cc0e64a2ab4 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 19 Jun 2025 18:03:35 +0200
Subject: [PATCH] clk: stm32: remove PCIE bus clock from the clock
 configuration

NOT_UPSTREAMABLE need to keep backward compatibility for old DT.

PCIE bus clock is now managed by SCMI, remove the clock from the clock
configuration.

Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: Ie61ab47be3ec298ebe4bd6e3556dd2fc419fc633
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/468570
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 drivers/clk/stm32/clk-stm32mp25.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

--- a/drivers/clk/stm32/clk-stm32mp25.c
+++ b/drivers/clk/stm32/clk-stm32mp25.c
@@ -336,7 +336,6 @@ enum enum_gate_cfg {
 	GATE_MDF1,
 	GATE_MSI_KER,
 	GATE_OSPIIOM,
-	GATE_PCIE,
 	GATE_PKA,
 	GATE_RNG,
 	GATE_RTCC3,
@@ -443,6 +442,7 @@ enum enum_gate_cfg {
 	GATE_MSI,
 	GATE_OSPI1,
 	GATE_OSPI2,
+	GATE_PCIE,
 	GATE_PLL1,
 	GATE_PLL2,
 	GATE_PLL3,
@@ -553,7 +553,6 @@ static const struct stm32_gate_cfg stm32
 	GATE_CFG(GATE_MCO2,		RCC_MCO2CFGR,		8,	0),
 	GATE_CFG(GATE_MDF1,		RCC_MDF1CFGR,		1,	0),
 	GATE_CFG(GATE_OSPIIOM,		RCC_OSPIIOMCFGR,	1,	0),
-	GATE_CFG(GATE_PCIE,		RCC_PCIECFGR,		1,	0),
 	GATE_CFG(GATE_PKA,		RCC_PKACFGR,		1,	0),
 	GATE_CFG(GATE_RTCC3,		RCC_C3CFGR,		26,	0),
 	GATE_CFG(GATE_RNG,		RCC_RNGCFGR,		1,	0),
@@ -663,6 +662,7 @@ static const struct stm32_gate_cfg stm32
 	GATE_CFG(GATE_MSI_KER,		RCC_D3DCR,		1,	0),
 	GATE_CFG(GATE_OSPI1,		RCC_OSPI1CFGR,		1,	0),
 	GATE_CFG(GATE_OSPI2,		RCC_OSPI2CFGR,		1,	0),
+	GATE_CFG(GATE_PCIE,		RCC_PCIECFGR,		1,	0),
 	GATE_CFG(GATE_PLL1,		RCC_PLL2CFGR1,		8,	0),
 	GATE_CFG(GATE_PLL2,		RCC_PLL2CFGR1,		8,	0),
 	GATE_CFG(GATE_PLL3,		RCC_PLL3CFGR1,		8,	0),
@@ -1364,12 +1364,6 @@ static struct clk_stm32_gate ck_icn_p_os
 	.hw.init = CLK_HW_INIT_INDEX("ck_icn_p_ospiiom", ICN_LS_MCU, &clk_stm32_gate_ops, 0),
 };
 
-/* PCIE */
-static struct clk_stm32_gate ck_icn_p_pcie = {
-	.gate_id = GATE_PCIE,
-	.hw.init = CLK_HW_INIT_INDEX("ck_icn_p_pcie", ICN_LS_MCU, &clk_stm32_gate_ops, 0),
-};
-
 /* PKA */
 static struct clk_stm32_gate ck_icn_p_pka = {
 	.gate_id = GATE_PKA,
@@ -2028,7 +2022,6 @@ static int stm32mp25_check_security(stru
 static const struct clock_config stm32mp25_clock_cfg[] = {
 	STM32_GATE_CFG(CK_BUS_ETH1, ck_icn_p_eth1, SEC_RIFSC(60)),
 	STM32_GATE_CFG(CK_BUS_ETH2, ck_icn_p_eth2, SEC_RIFSC(61)),
-	STM32_GATE_CFG(CK_BUS_PCIE, ck_icn_p_pcie, SEC_RIFSC(68)),
 	STM32_GATE_CFG(CK_BUS_ETHSW, ck_icn_p_ethsw, SEC_RIFSC(70)),
 	STM32_GATE_CFG(CK_BUS_ADC12, ck_icn_p_adc12, SEC_RIFSC(58)),
 	STM32_GATE_CFG(CK_BUS_ADC3, ck_icn_p_adc3, SEC_RIFSC(59)),
